js/parser/main.js

```diff
@@ -83,28 +84,50 @@ session.setTabSize(2);
 session.setUseSoftTabs(true);
 session.setUseWrapMode(false);
 
+// legacy url support
 var params = {};
 location.search.replace(/[?&](\w+)=([^&]*)/g, function(match, param, value){ params[param] = decodeURIComponent(value); });
 if ({}.hasOwnProperty.call(params, 'type') && (params.type === 'Module' || params.type === 'Script') && {}.hasOwnProperty.call(params, 'code')) {
-  (params.type === 'Script' ? radio : radio2).checked = true;
+  (params.type === 'Script' ? scriptRadio : moduleRadio).checked = true;
   session.setValue(params.code);
 }
 
+var params = {};
+location.search.replace(/[?&](\w+)=([^&]*)/g, function(match, param, value){
+  params[param] = decodeURIComponent(value);
+});
+if ('parse_type' in params) {
+  (params.parse_type === 'script' ? scriptRadio : moduleRadio).checked = true;
+}
+if ('script' in params) {
+  editor.setValue(params.script, -1);
+}
+
+var isLoaded = false;
 
 function onChange() {
   var code = editor.getValue();
-  var parseFn = radio.checked ? parser.parseScript : parser.parseModule;
+  var parseFn = scriptRadio.checked ? parser.parseScript : parser.parseModule;
   try {
     var ast = parseFn(code, { earlyErrors : true });
   } catch (ex) {
     displayError(ex);
     return;
   }
+
+  if (isLoaded) {
+    var args = {
```

Comment:
I would move this to its own function (`persistState`?), then add event listeners to the appropriate things. We could rename this function to something more appropriate, like `updateASTRendering`.

```js
editor.getSession().on('change', debounce(function(){
  updateASTRendering();
  persistState();
}, 300));

scriptRadio.addEventListener('change', updateASTRendering);
scriptRadio.addEventListener('change', persistState);

moduleRadio.addEventListener('change', updateASTRendering);
moduleRadio.addEventListener('change', persistState);

document.addEventListener('WebComponentsReady', updateASTRendering);
```

Now we no longer need `isLoaded`.